tools/crdb-physical-replication.sh

```diff
@@ -0,0 +1,237 @@
+    else   
+        echo "Creating certificate authority..."
+        cockroach cert create-ca --ca-key $primaryCApath --certs-dir $primarycertsdir
+    fi
+
+
+    # Look for existing certs
+    foundcount=0
+    for filename in "client.root.crt" "client.root.key" "node.crt" "node.key";
+    do
+        if [ -f $primarycertsdir/$filename ];
+        then
+            ((foundcount++))
+        fi
+    done
+
+    if [[ $foundcount -eq 4 ]]; # all there
+    then
+        read -p "Found existing certs in $primarycertsdir. Use those? (y=yes, n=you will need to delete them) " yn
+        if [ $yn != "y" ];
+        then
+            echo "You will need to delete or move the directory $primarycertsdir manually"
+            echo "After that, you can re-run this script"
+            exit
+        fi
+    fi
+
+    if [[ $foundcount -eq 0 ]];
+    then
+        echo "Creating node certificates..."
+        cockroach cert create-node localhost 127.0.0.1 "$(hostname -f)" ::1 localhost6 --certs-dir=$primarycertsdir --ca-key=$primaryCApath
+        echo "Creating client certificates..."
+        cockroach cert create-client root --certs-dir $primarycertsdir --ca-key $primaryCApath
+    fi
+ 
+    echo ""
+
+    listenaddr="localhost:9001"
+    httpaddr="localhost:8081"
+
+    echo "${bold}Starting Primary cluster (single node)...${normal}"
+    echo ""
+    cockroach start-single-node --certs-dir=$primarycertsdir --http-addr $httpaddr --listen-addr $listenaddr --config-profile replication-source --background
+    echo ""
+    echo "Your Standby cluster is running at $listenaddr, with the DB Console at $httpaddr"
+    echo ""
+    echo "This cluster will continue to run unless you terminate it manually"
+    echo ""
+ 
+    echo "Creating replication user..."
+    primaryurl="postgresql://root@$listenaddr/?options=-ccluster=system&sslmode=verify-full"
+
+    # pwgen is not on all systems, is there a better choice?
+    # https://unix.stackexchange.com/questions/230673/how-to-generate-a-random-string
+    primaryrepluserpass=$(openssl rand -base64 20)
+
+    sql="CREATE USER replication WITH LOGIN PASSWORD '$primaryrepluserpass';"
+    cockroach sql --execute "$sql" --url "$primaryurl" --certs-dir=$primarycertsdir
+
+    echo "Setting password..."
+    sql="ALTER USER replication WITH PASSWORD '$primaryrepluserpass';"
+    cockroach sql --execute "$sql" --url "$primaryurl" --certs-dir=$primarycertsdir
+
+    echo "Granting REPLICATION role..."
+    cockroach sql --execute "GRANT SYSTEM REPLICATION TO replication;" --url "$primaryurl" --certs-dir=$primarycertsdir
+
+    echo "Remember this connection string for use by the Standby cluster:"
+    primaryCAcertbytes="TODO" # read and url encode the CA bytes
+    urlforstandby="postgresql://replication:$primaryrepluserpass@$listenaddr/?options=-ccluster=system&sslmode=verify-full&sslinline=true&sslrootcert=$primaryCAcertbytes"
+    echo $urlforstandby
+    echo ""
+
+    echo "Starting workload... TODO"
```

Comment:
@knz Thoughts on a good way to run (authenticate) a workload? In particular, it looks like `workload` can’t take a `--certs-dir` parameter.